stream-api/src/main/kotlin/fund/cyber/markets/api/common/IncomingMessagesHandler.kt

```diff
@@ -37,9 +39,16 @@ class IncomingMessagesHandler(
                 }
             }
             is TradeChannelSubscriptionCommand -> {
-                tradesBroadcastersIndex.broadcastersFor(command.pairs, command.exchanges)
-                        ?.forEach { broadcaster -> broadcaster.registerChannel(wsChannel)
+                val broadcasters = tradesBroadcastersIndex.broadcastersFor(command.pairs, command.exchanges)
+                val result : LinkedList<Trade> = LinkedList()
+                while(result.size < 10 ) {
+                    val randomTrade = broadcasters.toMutableList()[Int.rand(0, broadcasters.size)]
+                            .getRandomTradeFromBroadcaster()
+                    val unique = result.none { it.tradeId == randomTrade.tradeId }
+                    if (unique) { result.add(randomTrade) }
                 }
+                WebSockets.sendText(jsonSerializer.writeValueAsString(result), wsChannel, null)
+                broadcasters?.forEach { broadcaster -> broadcaster.registerChannel(wsChannel) }
```

Comment:
Not sure if safe call (`?.`) needed there. As I see `broadcastersFor` returns non null collection.

Reply:
done